cpp/CMakeLists.txt

```diff
@@ -1,17 +1,25 @@
-cmake_minimum_required (VERSION 2.6)
+cmake_minimum_required (VERSION 3.6)
 
-project (grpc4bmi)
+project (grpc4bmi C CXX)
 set (GRPC4BMI_VERSION 0.2)
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/grpc4bmi.pc.cmake
                ${CMAKE_CURRENT_BINARY_DIR}/grpc4bmi.pc)
 
-add_compile_options(-std=c++11)
+if (NOT DEFINED CMAKE_CXX_STANDARD)
+  set (CMAKE_CXX_STANDARD 14)
+endif()
 
+find_package(Threads REQUIRED)
 find_package(Protobuf REQUIRED)
+message(STATUS "Using protobuf ${Protobuf_VERSION}")
 find_program(GRPC_CPP_PLUGIN grpc_cpp_plugin REQUIRED)
-find_library(GRPCPP_UNSECURE grpc++_unsecure)
-find_library(GRPCLIB grpc)
+find_library(GRPCREFLECTION grpc++_reflection)
+find_library(GRPCLIB grpc CONFIG REQUIRED)
+message(STATUS "Using gRPC ${GRPCLIB_VERSION}")
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(BMI REQUIRED IMPORTED_TARGET bmic bmicxx)
```

Comment:
Yep as described at https://github.com/eWaterCycle/grpc4bmi/blob/latest-protobuf/docs/server/Cpp.rst#installing-requirements. Instead of using git submodule to pin a version I use a tag during cloning.